wavedump/src/analyze-waveforms

```diff
@@ -90,134 +83,284 @@ def get_bins(x):
+            
             plt.xlabel("Charge (pC)")
-            plt.title(channel)
-
+            plt.legend()
+        if args.pdf:
+            prname=filename[:len(filename)-5]
+            plt.savefig("./FiguresAnalysis/"+prname+"_"+str(channel)+f"_{args.s}_{args.IT}.pdf")
         h.Write()
 
+    # The prefix `f_` will be used to denote objects that correspond to filtered data.
+        f_bins = get_bins(f_charge[channel])
+        f_h = ROOT.TH1D(f'f_{channel}', "Filtered Charge Integral for %s" % channel, len(f_bins), f_bins[0], f_bins[-1])
+        for x in f_charge[channel]:
+            f_h.Fill(x)
+        f_h.GetXaxis().SetTitle('Charge (pC)')
+        f_h.Write()
+        
+        # if args.plot:  # TODO: This part is only for testing
+        #     plt.figure()
+        #     bins = get_bins(average[channel]) 
+        #     plt.hist(average[channel],bins=bins,histtype='step',label='avg_'+channel)
+        #     plt.legend()
+        #
+        #     plt.figure()
+        #     plt.plot(np.arange(len(average[channel])), average[channel], label='avg_'+channel)
+        #     plt.legend()
+        #     num = 0
+        #     for x in average[channel]:
+        #         if x <= -35:
+        #             num += 1
+        #     print(f'num: {num}')
+        #
+        #     plt.figure()
+        #     bins = get_bins(std[channel])
+        #     plt.hist(std[channel],bins=bins,histtype='step',label='std_'+channel)
+        #     plt.legend()
+        #     print(len(std[channel]))
+        #
+        #     plt.figure()
+        #     plt.plot(np.arange(len(std[channel])), std[channel], label='std_'+channel)
+        #     plt.legend()
     f.Close()
 
-    for i in range(16,32):
-        data['ch_511'].append(None)
-        data['ch_511_rise_time'].append(None)
-        data['ch_511_fall_time'].append(None)
-
-    if args.upload:
-        if 'BTL_DB_HOST' not in os.environ:
-            print("need to set BTL_DB_HOST environment variable!",file=sys.stderr)
-            sys.exit(1)
-
-        if 'BTL_DB_PASS' not in os.environ:
-            print("need to set BTL_DB_PASS environment variable!",file=sys.stderr)
-            sys.exit(1)
-
-        print("Uploading results to the database...")
-        conn = psycopg2.connect(dbname='btl_qa',
-                                user='btl',
-                                host=os.environ['BTL_DB_HOST'],
-                                password=os.environ['BTL_DB_PASS'])
-        conn.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
-
-        cursor = conn.cursor()
-        cursor.execute("INSERT INTO btl_qa (barcode, voltage, ch_511, ch_511_rise_time, ch_511_fall_time, avg_pulse_x, avg_pulse_y, git_sha1, git_dirty, institution) VALUES (%(barcode)s, %(voltage)s, %(ch_511)s, %(ch_511_rise_time)s, %(ch_511_fall_time)s, %(avg_pulse_x)s, %(avg_pulse_y)s, %(git_sha1)s, %(git_dirty)s, 'Caltech'::inst)", data)
-        print("Done")
-
```

Comment:
So, for this database code, I guess in the end we will probably move this to the fitting code. I wonder now if it's better just to combine the two scripts.

In any case, we need to save this code as it's the code which writes the results to the database.